scripts/apps/authoring/authoring/directives/ItemCarouselDirective.js

```diff
@@ -117,12 +121,36 @@ export function ItemCarouselDirective($timeout) {
             });
 
             elem.on('drop dragdrop', (event) => {
+                const type = getSuperdeskType(event);
+
                 event.preventDefault();
                 event.stopPropagation();
 
-                ctrl.initializeUploadOnDrop(scope, event);
+                if (ALLOWED_TYPES.includes(type) || type === 'Files') {
+                    ctrl.initializeUploadOnDrop(scope, event);
+                } else {
+                    const allowedTypeNames = [
+                        (scope.allowPicture === true ? gettext('image') : ''),
+                        (scope.allowVideo === true ? gettext('video') : ''),
+                        (scope.allowAudio === true ? gettext('audio') : ''),
+                    ].filter(Boolean).join(', ');
+                    const message = gettext('Only the following content item types are allowed: ');
+
+                    notify.error(message + allowedTypeNames);
+                }
             });
 
+            /**
+             * Get superdesk type for data transfer if any
+             *
+             * @param {Event} event
+             * @return {string}
+             */
+            function getSuperdeskType(event) {
+                return event.originalEvent.dataTransfer.types
+                    .find((name) => name.includes('application/superdesk') || name === 'Files');
```

Comment:
It is a type accepted by superdesk. Probably we can rename the function. Do you have a proposal for a better name?

Reply:
I changed the description of the function

Reply:
looks like we have this kind of function in [other](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/authoring/authoring/controllers/AssociationController.ts#L79) [places](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/apps/relations/directives/RelatedItemsDirective.ts#L106) would be good to avoid this duplication and move it to some core file

Reply:
done